src/testRunner/unittests/tsserver/projects.ts

```diff
@@ -1481,6 +1481,48 @@ namespace ts.projectSystem {
             ]);
         });
 
+        it("synchronizeProjectList returns correct information when base configuration file cannot be resolved", () => {
+            const file: File = {
+                path: `${tscWatch.projectRoot}/index.ts`,
+                content: "export const foo = 5;"
+            };
+            const config: File = {
+                path: `${tscWatch.projectRoot}/tsconfig.json`,
+                content: JSON.stringify({ extends: "./tsconfig_base.json" })
+            };
+            const host = createServerHost([file, config, libFile]);
+            const projectService = createProjectService(host);
+            projectService.openClientFile(file.path);
+            const knownProjects = projectService.synchronizeProjectList([], /*includeProjectReferenceRedirectInfo*/ false);
+            assert.deepEqual(knownProjects[0].files, [
+                libFile.path,
+                file.path,
+                config.path,
+                `${tscWatch.projectRoot}/tsconfig_base.json`,
+            ]);
+        });
+
+        it("synchronizeProjectList returns correct information when base configuration file cannot be resolved and redirect info is requested", () => {
+            const file: File = {
+                path: `${tscWatch.projectRoot}/index.ts`,
+                content: "export const foo = 5;"
+            };
+            const config: File = {
+                path: `${tscWatch.projectRoot}/tsconfig.json`,
+                content: JSON.stringify({ extends: "./tsconfig_base.json" })
+            };
+            const host = createServerHost([file, config, libFile]);
+            const projectService = createProjectService(host);
+            projectService.openClientFile(file.path);
+            const knownProjects = projectService.synchronizeProjectList([], /*includeProjectReferenceRedirectInfo*/ true);
+            assert.deepEqual(knownProjects[0].files, [
+                { fileName: libFile.path, isSourceOfProjectReferenceRedirect: false },
+                { fileName: file.path, isSourceOfProjectReferenceRedirect: false },
+                { fileName: config.path, isSourceOfProjectReferenceRedirect: false },
+                { fileName: `${tscWatch.projectRoot}/tsconfig_base.json`, isSourceOfProjectReferenceRedirect: false },

```

Comment:
Why include it instead of removing it from the response? Are there other cases where TSServer could tell us to expect a file that doesn't actually exist?

Reply:
That is, where we're not expected to then create it? Of course, move to file probably gives a non-existent file path but expects us to go create it - I'm not sure about that though.

Reply:
I am not sure. but i am certain that extendedSourceFiles should contain source file name that wasnt present. It would need another file exists check or another field just to return information if file is present or not and that seems unnecessary just to let project structure information. Let me know if you think its necessary. If it is we would probably need another field on tsconfig source files

Reply:
Ping @uniqueiniquity

Reply:
Sorry for the delay.
After further thought, it's probably fine.
If you send VS a file here that doesn't exist, the user will see a somewhat strange diagnostic in the error list complaining that the file doesn't exist, but it won't break or crash. 
If we ask for semantic diagnostics for the tsconfig that does exist, will we get one complaining about the missing referenced file?

Reply:
thats correct